### PR TITLE
newest version of task list layout and behaviour

### PR DESCRIPTION
**Pull request by bangyii**

### CODE REVIEW

File: client/src/components/TaskList/TaskList.js [hunk 1]
```diff
@@ -45,6 +63,16 @@ export default function TaskList() {
     const dragEndHandler = (result) => {
         //If result is invalid, return
         if (!result.destination || result.destination.index === result.source.index || !result) {
+            console.log("Invalid", result, toDelete.current);
+            // if (result.source.index !== undefined && toDelete.current) {
```
Comment (NganJason) [inline client/src/components/TaskList/TaskList.js:67]: do we still need this?

File: client/src/components/Task/Task.js [hunk 2]
```diff
@@ -67,38 +61,51 @@ export default function Task(props) {
...
         if (dragging) {
             setOptionsVisible(false);
-            setEditingContent(false);
         }
-    }, [dragging])
+    }, [dragging]);
+
+    //Effect when main store content is changed
+    React.useEffect(() => {
+        setTemporaryContent({content: content.content, lastEdit: new Date().getSeconds()});
```
Comment (NganJason) [inline client/src/components/Task/Task.js:69]: Just curious, when will mainContent.lastEdit > temporaryContent.lastEdit

correct me if im wrong, so even if we update the temporaryContent instantly after the mainContent is updated, there will still be a short period of time whereby the temporaryContent.lastEdit will be < mainContent.lastEdit?

File: client/src/components/TaskList/TaskList.js (hunk 1 above)
Comment (bangyii) [reply client/src/components/TaskList/TaskList.js:67]: Part of the implementation for drag to delete, not done properly yet. I wanna just leave it first

File: client/src/components/Task/Task.js (hunk 2 above)
Comment (bangyii) [reply client/src/components/Task/Task.js:69]: mainContent > temporaryContent for the single rerender immediately after dragEnd happens.

There's no way for us to "immediately" update temporaryContent with the current structure, the earliest we can do is 1 frame after mainContent was updated. So without lastEdit, we will have 1 wrong frame.

File: client/src/components/TaskList/TaskList.js [hunk 3]
```diff
@@ -3,23 +3,41 @@ import React from "react";
...
+                //Scroll to end of task list
+                const elem = document.getElementById("task-list-paper");
+                elem.scrollTop = elem.scrollHeight;
+            }, 0);
+
+            return newContents;
+        });
+    }
+
     React.useEffect(() => {
```
Comment (NganJason) [inline client/src/components/TaskList/TaskList.js:41]: @bangyii  why do we need this when we already have line25-44 in PlayPauseButton.js?

**Pull request merged**